python/ray/serve/scripts.py

```diff
@@ -323,6 +324,16 @@ def deploy(config_file_name: str, address: str):
         "as the ingress deployment."
     ),
 )
+@click.option(
+    "--reload",
+    is_flag=True,
+    help=(
+        "Listens for changes to files in the working directory, --working-dir "
+        "or the working_dir in the --runtime-env, and automatically redeploys "
+        "the application. This will block until Ctrl-C'd, then clean up the "
+        "app. The --blocking/--non-blocking flag is ignored."
```

Comment:
Instead of ignoring, can we raise an explicit exception when `--non-blocking` is provided?

Reply:
I have added raising a `click.ClickException` if `reload` is True and `blocking` is False.

The message is:

>The --non-blocking option conflicts with the --reload option.

I am open to different wording.